cmake/cores/Cortex-A9.cmake

```diff
@@ -4,55 +4,48 @@
 set(CMAKE_SYSTEM_PROCESSOR cortex-a9)
 
 if(MBED_TOOLCHAIN STREQUAL "GCC_ARM")
-
-set(GCC_FLAGS " \
-    -mthumb-interwork \
-    -marm \
-    -march=armv7-a \
-    -mfpu=vfpv3 \
-    -mfloat-abi=hard \
-    -mno-unaligned-access \
-")
-
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} \
-    ${GCC_FLAGS} \
-")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} \
-    ${GCC_FLAGS} \
-")
-set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} \
-    ${GCC_FLAGS} \
-")
-set(CMAKE_CXX_LINK_FLAGS "${CMAKE_CXX_LINK_FLAGS} \
-    ${GCC_FLAGS} \
-")
-
+    list(APPEND common_toolchain_options
+        "-mthumb-interwork"
+        "-marm"
+        "-march=armv7-a"
+        "-mfpu=vfpv3"
+        "-mfloat-abi=hard"
+        "-mno-unaligned-access"
+    )
+
+    target_compile_options(mbed-os
+        PUBLIC
+            ${common_toolchain_options}
+    )
+
+    target_link_options(app
```

Comment:
Core cmake scripts should avoid influencing link options for the app directly. Instead, it should set link options to `mbed-os` target with PUBLIC keyword so that cmake can propagate the link option to the consumer (app).
Same comment for other cores script changes as well.

Reply:
You cannot use `target_link_options()` to set link options for an object library. `mbed-os` is built as an object library.

Reply:
Working on it. I initially thought that the command could only be used for shared libraries because it did not work. I must have something else wrong at the time. I will push something to update the first commit in the branch.

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/69b9a15a7753f73fe4744211d1bb895ebadc39a1..82358905db82d7c7d76c4cb49ccfe2a1eba87a95).